master/getting-started/kubernetes/upgrade/delete.md

```diff
@@ -3,5 +3,68 @@ title: Deleting old data
 no_canonical: true
 ---
 
-_Docs on this coming soon!_
+## About deleting old Calico data
 
+You may need or wish to manually delete Calico data from your etcd datastore under the
+following conditions.
+
+- [A data migration attempt failed partway through, leaving the etcdv3 datastore
+  with some, but not all of your etcvd2 data](#deleting-calico-data-from-etcdv3-after-a-partial-migration).
+  
+- [You succeeded in migrating your data and upgrading to Calico v3.0. After
+  running Calico for some time and experiencing no errors, you want to delete
+  the Calico data from the etcdv2 datastore](#deleting-calico-data-from-etcdv2-after-a-successful-migration-and-upgrade).
+  
+
+## Deleting Calico data from etcdv3 after a partial migration
+
+1. Obtain the name of the etcd pod. 
+
+   ```
+   kubectl get pods --all-namespaces
+   ```
+
+1. Copy the name of the etcd pod.
+
+1. Use the following command to open an interactive shell inside your etcd pod.
+
+   ```
+   kubectl exec -it <etcd-pod-name> -n kube-system -- /bin/sh
+   ```
+   
+1. Issue the following command to retrieve a list of all of the Calico keys.
+   
+   ```
+   ETCDCTL_API=3 etcdctl --endpoints=http://localhost:6666 get / --prefix --keys-only
+   ```
+   
+   > **Note**: If you are not on a kubeadm cluster, you can remove 
+   > `--endpoints=http://localhost:6666` from this and the remaining commands.
+   {: .alert .alert-info}
+   
+   If this command does not return a series of `/calico/` keys, you may not
+   be in the etcd instance that's used by {{site.prodname}}. Return to step 1 and try a
+   different etcd pod.
+   
+1. Issue the following command to delete the {{site.prodname}} keys.
+   
+   ```
+   ETCDCTL_API=3 etcdctl --endpoints=http://localhost:6666 del / --prefix 
+   ```
+   
+   It returns the number of keys it deleted.
+   
+1. Issue the following command to confirm that the {{site.prodname}} keys were deleted.
+   
+   ```
+   ETCDCTL_API=3 etcdctl --endpoints=http://localhost:6666 get / --prefix --keys-only
+   ```
+   
+   It should return nothing.
+   
+1. Congratulations! You've cleaned {{site.prodname}}'s etcdv3 datastore of {{site.prodname}}
+   data. 
+   
+## Deleting Calico data from etcdv2 after a successful migration and upgrade
```

Comment:
I'd put this one at the top since it should be the mainline case that most people care about.